### PR TITLE
Enable gateway protocols with interface-level granularity

### LINKED ISSUE #1413: [BUG] Global and node level 'gateway' attributes not propagated to interfaces
### Describe the bug
In the given topology, the intent is to make ```n1``` a gateway on VLAN v1

### To Reproduce
```netlab create test.yml```

### Expected behavior
n1 should be configured as VRRP gateway

### Lab topology

```
module: [gateway,vlan]
gateway.protocol: vrrp

vlans:
 v1:

nodes:
 n1:
  vlans:
   v1:
    gateway: True
```

### Output
```
  File "/home/jvanbemmel/netlab/netsim/modules/gateway.py", line 39, in check_protocol_support
    gw_proto = intf.gateway.protocol                                  # We're asuming someone else did a sanity check on this value
               ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'bool' object has no attribute 'protocol'

```

### Version

dev branch

### Additional context
Suggested fix, add in gateway.py ```check_protocol_support``` line 36
```
    if intf.gateway is True:                                          # Expand global and node specific gateway parameters
      intf.gateway = topology.gateway + node.get('gateway',{})
```

### PR DESCRIPTION
**Pull request by ipspace**
Fixes #1413, replaces #1414

### CODE REVIEW

**Comment by ipspace** [comment]
Documentation is coming in a separate commit ;)

File: netsim/modules/gateway.py [hunk 1]
```diff
@@ -153,8 +153,50 @@ def module_init(self, topology: Box) -> None:
...
+    copy_to_link = False
+    for intf in link.interfaces:
+      if 'gateway' not in intf:                             # No 'extra special' stuff, move on
+        continue
+      if intf.get('gateway',None) is False:                 # The 'special' stuff is don't do it?
+        continue                                            # ... sure, I can do that ;)
+      if 'gateway' in link:                                 # Do we already have 'gateway' on link?
+        if link.gateway is False:                           # If we have 'gateway:false' we're dealing with a stupidity
+          log.error(
+            'Cannot combine gateway:false on link with gateway:true on interface',
```
Comment (jbemmel) [inline netsim/modules/gateway.py:175]: ```gateway``` on interface could also be a dict setting the vrrp.priority for example

"Cannot combine gateway:false on link with gateway attribute on interface" (although I could come up with scenarios where that could make sense to me, I'll spare you the eye-rolling exercise this time)

**Comment by jbemmel** [comment]
```
---
module: [gateway,vlan]

gateway:
 protocol: vrrp
 id: 1

vlans:
 core:
 access:
  # gateway: True -> This enables the gateway

groups:
 _auto_create: True
 
 core:
  members: [ c1, c2, c3 ]
  vlans:
   core:
    gateway: True # -> This does work
 
 access:
  members: [ a1, a2 ]
  vlans:
   access:
    gateway: True  # -> This does not

 hosts:
  members: [ h1 ]
  device: linux

links:
- c1:
  c2:
  c3:
  a1:
  a2:
  vlan.trunk: [ core ]

- a1:
    vlan.access: access
  a2:
    vlan.access: access
  h1:
```

Possibly not an issue introduced by this PR, but enabling the gateway in a trunk vlan works but not in an access vlan

**Comment by ipspace** [comment]
> Possibly not an issue introduced by this PR, but enabling the gateway in a trunk vlan works but not in an access vlan

Nice one, thank you. Will figure out what's going on. Merge postponed.

File: netsim/modules/gateway.py (hunk 1 above)
Comment (jbemmel) [reply netsim/modules/gateway.py:175]: ```Warning: interface gateway attributes ignored due to link.gateway=False```

**Comment by ipspace** [comment]
> Possibly not an issue introduced by this PR, but enabling the gateway in a trunk vlan works but not in an access vlan

That issue is related to #1424. It looks like I copied node vlan data into interface data only on trunks but not on access links. Will open a separate issue for that and fix the code.

The "this should be a warning" part has been implemented, merging...

**Pull request merged**